docs/DesignDocs/SGX_QuoteEx_Integration.md

```diff
@@ -0,0 +1,121 @@
+# Attestation: OE SDK Integration with Intel® SGX SDK quote-ex Library for Evidence Generation in New Formats
+
+This design document proposes an extension of the OE SDK implementation for integration with Intel® SGX SDK quote-ex library, for support of generation of evidence in new SGX formats such as Enhanced Privacy ID (EPID).
+
+# Motivation
+
+Existing implementation of OE SDK SGX attestation, based on Intel® SGX SDK Data Center Attestation Primitives (DCAP) quote generation library (simply called the DCAP library), only supports generation of evidence in a single SGX ECDSA-p256 evidence format.
+
+On some SGX platforms, other evidence formats - including those based on Enhanced Privacy ID (EPID), are supported and preferred by some application solutions. Generation of evidence in these formats is supported by the Intel® SGX SDK with a library package, libsgx-quote-ex (or simply called quote-ex).
+
+# User Experience
+
+The proposed extension only changes the internal implementation of the OE SDK attestation software stack. It does not impact the OE SDK API as described in [PR #2621](https://github.com/openenclave/openenclave/pull/2621). With the integration of the quote-ex library, an attester application enclave's call to OE SDK API `oe_sgx_get_attesters()` returns the list of all SGX evidence attester plugins available to the calling enclave instance.
```

Comment:
Just FYI, I've commented in PR 2621 that I don't think the oe_sgx_get_attesters api should be named that, so if that is renamed then this line would need to updated at that time.

Reply:
Yes, this document will stay in sync with PR 2621 on API definition.